pkg/config/merge.go

```diff
@@ -0,0 +1,16 @@
+package config
+
+// extendedService is merged into service
+func merge(service *Service, extendedService *Service) {
+	// rules based on https://docs.docker.com/compose/extends/#adding-and-overriding-configuration
+	mergeMap(service.Environment, extendedService.Environment)
+	if len(extendedService.Image) > 0 {
+		service.Image = extendedService.Image
+	}
+}
+
+func mergeMap(env map[string]string, extendedEnv map[string]string) {
```

Comment:
Rename to mergeStringMap and rename args

Reply:
done